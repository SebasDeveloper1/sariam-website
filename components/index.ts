import Button from './Buttons/Button/Button';
import IconButton from './Buttons/IconButton/IconButton';
import SliderComponent from './SliderComponent/SliderComponent';
import ServiceCard from './ServiceCard/ServiceCard';
export { Button, IconButton, SliderComponent, ServiceCard };
