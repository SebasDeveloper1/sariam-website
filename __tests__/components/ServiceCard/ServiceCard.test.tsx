import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ServiceCard } from '@/components';
import { mockServiceProps } from '@/containers/ServicesSection/servicesSection.model';

const mockService: mockServiceProps = {
  sys: { id: '2QIOIAuoSMwGRRvYqP7Tyy' },
  title: 'Logística y Servicio',
  cardDescription: 'Logística y personal de servicio.',
  generalDescription:
    'Experimenta un evento sin preocupaciones con nuestro servicio de logística y personal de atención. Desde la planificación hasta la ejecución, nuestro equipo profesional y altamente capacitado se encarga de cada detalle. Brindamos una experiencia de primera clase, atendiendo las necesidades de tus invitados con excelencia.',

  serviceDetails: [
    'Coordinador logístico.',
    'Meseros.',
    'Bartenders.',
    'Cocineros.',
  ],

  media: {
    title: 'logistic1',

    description: 'Video por Nacho Producciones.',

    url: 'https://videos.ctfassets.net/h5onrwafbbzu/1R92JNkSlUpBDdmyHMlwRF/a6d50e63aecc95d0583a52fc1c852487/food.mp4',
  },

  icon: <div id="mock-icon">Icon</div>,

  bgCard: 'bg-thumbnail-yellow',
};

let card: HTMLElement;

describe('ServiceCard component', () => {
  // Before running the tests, render the ServiceCard component with the mockService data
  beforeAll(() => {
    const { getByRole } = render(<ServiceCard data={mockService} />);
    card = getByRole('article');
  });

  describe('Rendering', () => {
    // Test that the ServiceCard component is rendered
    test('should render ServiceCard component', () => {
      expect(card).toBeInTheDocument();
    });

    // Test that ServiceCard's title is rendered correctly
    test("Should render ServiceCard's title", () => {
      expect(card.querySelector('h3')).toHaveTextContent(mockService.title);
    });

    // Test that ServiceCard's description is rendered correctly
    test("Should render ServiceCard's description", () => {
      expect(card.querySelector('p')).toHaveTextContent(
        mockService.cardDescription
      );
    });
  });

  describe('Snapshot', () => {
    // Test that the rendered ServiceCard matches the snapshot
    test('matches snapshot', () => {
      const { asFragment } = render(<ServiceCard data={mockService} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
