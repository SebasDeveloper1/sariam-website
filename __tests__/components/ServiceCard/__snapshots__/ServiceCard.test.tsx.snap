// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard component Snapshot matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden col-span-6 md:col-span-3 xl:col-span-2 w-full h-full min-h-[7rem] rounded-2xl hover:bg-gray-700/50 cursor-pointer"
    role="article"
    tabindex="0"
  >
    <div
      class="w-full h-full bg-thumbnail-yellow bg-no-repeat bg-right bg-contain"
    >
      <div
        class="relative flex justify-center items-center gap-5 w-full h-full p-6 text-start bg-gray-900/25"
      >
        <span
          class="flex items-center justify-center aspect-square rounded-full bg-lime-50 text-5xl text-lime-500"
        >
          <div
            id="mock-icon"
          >
            Icon
          </div>
        </span>
        <div
          class="w-full space-y-1"
        >
          <h3
            class="text-xl font-semibold text-white"
          >
            Logística y Servicio
          </h3>
          <p
            class="text-sm text-white"
          >
            Logística y personal de servicio.
          </p>
        </div>
        <span
          class="absolute top-3 right-3 text-2xl text-white"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 256 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
